modulus/models/fcn_mip_plugin.py

```diff
@@ -159,21 +159,25 @@ def __init__(
     def prepare_input(self, input, time):
         device = input.device
         dtype = input.dtype
-        num_chans = input.size(2)
-        input_list = list(torch.split(input, 1, dim=1))
+
         i = self.input_map_wts.row.values - 1
         j = self.input_map_wts.col.values - 1
         data = self.input_map_wts.S.values
-        M = torch.sparse_coo_tensor(np.array((i, j)), data).to(device).type(dtype)
-
-        for i in range(len(input_list)):
-            input_list[i] = input_list[i].reshape(num_chans, -1) @ M.T
-            input_list[i] = input_list[i].reshape(1, num_chans, 6, 64, 64)
+        M = torch.sparse_coo_tensor(np.array((i, j)), data).type(dtype).to(device)
 
+        input = input.reshape(input.shape[0], input.shape[1], input.shape[2], -1) @ M.T
```

Comment:
I found a bug while using Pytorch 1.13.1 that we should consider:

```python 
input.reshape(input.shape[0], input.shape[1], input.shape[2], -1) @ M.T
```

gives the error

```
NotImplementedError: Could not run 'aten::sparse_dim' with arguments from the 'CUDA' backend
```

while

```python
temp = input_list[i].reshape(num_ens * num_chans, -1)
temp = (M @ temp.T).T
```

does not. This is likely due to some limitation in matrix batch multiplication for sparse tensors in Pytorch, Consider revising here and in `prepare_output`.

Reply:
Great comment. I added a fix that just reshapes the data to a 2d array and then changes the shape again after multiplication. Can you have a look again?